src/hrc1155.ts

```diff
@@ -0,0 +1,45 @@
+import { Harmony } from '@harmony-js/core'
+import { BigNumber, logger } from 'ethers'
+import { Logger } from 'ethers/lib/utils'
+import { BaseToken } from './base-implementation'
+
+export class HRC1155 extends BaseToken {
+  constructor(address: string, abi: any, client: Harmony) {
+    super(address, abi, client)
+  }
+
+  async balanceOf(address: string, id: string): Promise<number> {
+    if (!address) {
+      throw new Error('You have provide an address')
+    }
+
+    try {
+      const balance: BigNumber = await this._getBalance(address, id)
+      return balance.toNumber()
+    } catch (error) {
+      return logger.throwError('bad result from backend', Logger.errors.SERVER_ERROR, {
+        method: 'balanceOf',
+        params: address,
+        error,
+      })
+    }
+  }
```

Comment:
let's make `_getBalance` return a number directly. This method should only do `return this._getBalance` and leave the error handling to the underlying method

Reply:
Okay, I'll fix it!